src/pipeline.cc

```diff
@@ -413,6 +414,42 @@ void Quit(SemaManager &manager) {
   no_active_threads.wait(lock, [] { return !active_threads; });
 }
 
+struct HeadersData {
+  std::size_t content_length;
+};
+
+void ReadHeaders(bool* eof, HeadersData* headers) {
+  static const std::string_view kContentLength("Content-Length: ");
+  std::string buf;
+  while (true) {
+    int c = getchar();
+    if (c == EOF) {
+      *eof = true;
+      return;
+    }
+    if (c == '\n') {
+      if (buf.empty()) {
+        if (headers->content_length == 0)
+          LOG_S(WARNING) << "Content-Length was not defined in header";
+        return;
+      }
+      if (!buf.compare(0, kContentLength.size(), kContentLength)) {
+        std::from_chars_result result = std::from_chars(
```

Comment:
Unclear why the plain old atoi should be replaced by the much more complicated from_chars...